packages/config/src/handlers/forms/validation.ts

```diff
@@ -0,0 +1,424 @@
+  mother: [
+    'exactDateOfBirthUnknown',
+    'ageOfIndividualInYears',
+    'iD',
+    'motherNidVerification',
+    'maritalStatus',
+    'multipleBirth',
+    'occupation',
+    'educationalAttainment'
+  ],
+  father: [
+    'exactDateOfBirthUnknown',
+    'ageOfIndividualInYears',
+    'iD',
+    'fatherNidVerification',
+    'maritalStatus',
+    'occupation',
+    'educationalAttainment'
+  ],
+  deceased: [
+    'exactDateOfBirthUnknown',
+    'ageOfIndividualInYears',
+    'deceasedID',
+    'maritalStatus'
+  ],
+  deathEvent: [
+    'mannerOfDeath',
+    'causeOfDeathEstablished',
+    'causeOfDeathMethod',
+    'deathDescription'
+  ],
+  marriageEvent: ['typeOfMarriage'],
+  groom: [
+    'exactDateOfBirthUnknown',
+    'ageOfIndividualInYears',
+    'iD',
+    'marriedLastNameEng'
+  ],
+  bride: [
+    'exactDateOfBirthUnknown',
+    'ageOfIndividualInYears',
+    'iD',
+    'marriedLastNameEng'
+  ],
+  informant: [
+    'registrationPhone',
+    'registrationEmail',
+    'exactDateOfBirthUnknown',
+    'ageOfIndividualInYears',
+    'informantID',
+    'informantNidVerification',
+    'primaryAddressSameAsOtherPrimary'
+  ],
+  witnessOne: [],
+  witnessTwo: []
+}
+
+const form = z.object({
+  sections: z
+    .array(
+      section
+        .refine(
+          (sec) =>
+            findDuplicates(
+              sec.groups.flatMap(({ fields }) => fields.map(({ name }) => name))
+            ).length === 0,
+          (sec) => ({
+            message: `Field names in a section should all be unique. Duplicate field(s): ${findDuplicates(
+              sec.groups.flatMap(({ fields }) => fields.map(({ name }) => name))
+            ).join(', ')} found`
+          })
+        )
+        .refine(
+          (sec) => {
+            const fieldsInSection = sec.groups.flatMap((group) =>
+              group.fields.map(({ name }) => name)
+            )
+            return (
+              (REQUIRED_FIELDS_IN_SECTION[sec.id]?.filter(
+                (reqField) => !fieldsInSection.includes(reqField)
+              ).length ?? 0) === 0
+            )
+          },
+          (sec) => {
+            const fieldsInSection = sec.groups.flatMap((group) =>
+              group.fields.map(({ name }) => name)
+            )
+            const notIncludedRequiredFields = (
+              REQUIRED_FIELDS_IN_SECTION[sec.id] ?? []
+            ).filter((reqField) => !fieldsInSection.includes(reqField))
+            return {
+              message: `Missing required fields "${notIncludedRequiredFields.join(
+                ', '
+              )}" found`
+            }
+          }
+        )
+        .refine(
+          (sec) => {
+            if (!REQUIRED_FIELDS_IN_SECTION[sec.id]) {
+              return true
+            }
+            const nonCustomfieldsInSection = sec.groups.flatMap((group) =>
+              group.fields
+                .filter(({ custom }) => !Boolean(custom))
+                .filter(({ type }) => type !== 'DIVIDER')
```

Comment:
I think we need to filter all the `viewOnly` fields here

Reply:
OK.  Thats probably a minor hotfix if we get around to it